### PR TITLE
Summary endpoint for individual contribution totals

### LINKED ISSUE #1973: Create summary endpoint for location of individual contributions
So that users learn how much money comes from individuals in different states, create an endpoint that shows the total amount raised via individual contributions from each state for various types of groups. 

---

This issue is for creating the endpoint that will serve the data for this visualization:
map

For more details, check out 

**Completion criteria (originally from [this comment](https://github.com/18F/openFEC/issues/1620#issuecomment-218630091)):**
- [ ] The endpoint returns the _total_ amount raised via individual contributions by all groups combined from individuals from a certain state
- [ ] The endpoint returns the total amount raised via individual contributions by each of these kinds of groups:
  - All candidates
  - Presidential candidates
  - Senate candidates
  - House candidates
  - All PACs
  - Super PACs
  - SSFs
  - Non-connected
  - Leadership PACs
  - Hybrid PACs
  - National party committees

### PR DESCRIPTION
**Pull request by ccostino**
Closes #1973 

This changeset introduces a new endpoint for viewing all of the individual summary contribution totals based on Schedule A and Committee data broken down by state, committee type, and cycle.  This new endpoint contains the following additions:

* A new materialized view, `ofec_sched_a_aggregate_state_recipient_totals_mv` (derived from data found in `ofec_sched_a_aggregate_state` and `ofec_committee_detail_mv`)
* A new model, `ScheduleAByStateRecipientTotals`
* A new view, `ScheduleAByStateRecipientTotalsView`
* A new schema, `ScheduleAByStateRecipientTotalsSchema` (and an associated paging schema)
* A new resource endpoint, `/schedules/schedule_a/by_state/totals/`, registered with the API docs
* Several new tests focused on the new endpoint and model

The endpoint does provide a couple of additional pieces of information not needed for the charts per se, but could be useful as metadata for someone accessing the API directly.  The resource also allows you to filter on multiple states, committee types, and cycles in addition to sorting on these columns.

**For reviewers:**

As this is the first endpoint I've built, I have several questions around what I've built:

* Is the SQL correct in construction the `ofec_sched_a_aggregate_state_recipient_totals_mv`, and is the math correct for calculating the totals and counts?  The counts in particular I'm curious about, especially the totals of totals counts.
* Is the view construction correct?  I think I have the sort arguments and paging working correct, but would like to know for sure.
* Could the sorting be improved at all?  I know we only support sorting on a single column, but for this endpoint I feel like it'd really help supporting more than one.  Worth considering?
* There must be a better way to setup those tests...

I've tried to keep things as simple as possible, but I also wonder if I missed anything.  If so, please let me know!

/cc @jontours, @LindsayYoung, @noahmanger, @xtine

### CODE REVIEW

**Comment by LindsayYoung** [comment]
I can take a look at this

File: data/sql_updates/sched_a_by_state_recipient_totals.sql [hunk 1]
```diff
@@ -0,0 +1,90 @@
+-- Schedule A summary of individual contributions by state raised for each committee type per cycle.
+drop table if exists ofec_sched_a_aggregate_state_recipient_totals_tmp;
```
Comment (ccostino) [inline data/sql_updates/sched_a_by_state_recipient_totals.sql]: No worries!  I will adjust, thanks for catching that!

File: data/sql_updates/sched_a_by_state_recipient_totals.sql [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            --    ofec_fips_states
+        )
+    group by
+        agg_st.cycle,
+        agg_st.state,
+        agg_st.state_full,
+        cd.committee_type,
+        cd.committee_type_full
+),
+overall_total as (
```
Comment (ccostino) [inline data/sql_updates/sched_a_by_state_recipient_totals.sql]: D'oh!  Thanks for the catch, adding two more groupings to account for these totals.

**Comment by LindsayYoung** [comment]
Almost good to go:
- add sort options for all_pacs, all_candidates and all
- update docs

**Comment by ccostino** [comment]
There is still a bug to iron out with this; supplying `all`, `all_candidates`, and `all_pacs` as a `committee_type` is returning 0 results despite them being in the materialized view.

**Comment by ccostino** [comment]
This is likely due to the fact that these values in the database need to be upper case and right now they're not in the script.  Will fix first thing tomorrow.

**Comment by ccostino** [comment]
Bug fixed as it was bugging me too much...

**Pull request merged**